### ISSUE TITLE
Download new quests automatically

### ISSUE THREAD

**Issue by pietervdvn** [westnordost/StreetComplete#699]
Hey,

First of all, thanks for the amazingly usefull app! You're not far away of 1 million edits.

A small usability error that I encountered was the not automatic updating of quests. 

I scanned for quests once. In the mean time, I've fixed a lot of road surfaces in parallel, with another editor. However, the 'What is this road surface?' quests were still shown in StreetComplete, even weeks later. That was until I realized what happened and rescanned for quests on my entire city.

This could be more polished; e.g. by auto updating the quests every once in a while. I would propose that quests would 
- Attempt to update daily for 'wifi-only' and 'constant syncing'-users, for a bounding box of around one or two kilometer around the user.
- Be redownloaded daily ẁhen manually syncing for `manual` users.

**Comment by HolgerJeromin** [comment]
Thanks for your report. The same is handled here:  #562

**Comment by westnordost** [comment]
Basically, that is what is done already, but the interval is not set to daily, but to 2 weeks, iirc. However, if the app notices that it has outdated data (when it is trying to upload an answer and gets a conflict because the same info has already been supplied by someone else / another editor), it will automatically update the area.

Until data has been invalidated (after 2 weeks or if it ran into a conflict), the data will not update no matter how many times you hit the download button. The reason for that relatively long interval is that I do not want this app to put a big load on the overpass api, which it heavily uses.

**Comment by pietervdvn** [comment]
For my usage, 2 weeks is to long...

I understand your politeness towards overpass. Otho, you have 8000 users;
whereas overpass allows up to 10.000 queries a day a project... That means
that each of your users is allowed a single query daily.

For most users, two weeks is good; but an option offering more regular
updates would help me.

Met vriendelijke groeten,
Pieter Vander Vennet

**Comment by pietervdvn** [comment]
Hey,

I dove in the source code.

The quests expire after one week (not two). My custom build allows 15
minutes as refresh rate now ;)

I edited the user agent as well, so if I get banned from overpass, the rest
of Surrey complete isn't.

Btw: excellent code quality! I found my way around quite well

Met vriendelijke groeten,
Pieter Vander Vennet

**Comment by westnordost** [comment]
> Btw: excellent code quality! I found my way around quite well

Thank you, I appreciate this very much :-)

So, it would be possible / okay to add an option in the advanced part of the settings that would invalidate the whole quest cache. I'd accept a PR that adds that.

Regarding Overpass, note that for each single download of a quest in one area, at least one overpass query is made. So it is not only one query per user per day if the quests were invalidated after one day.

**Comment by ENT8R** [comment]
> So, it would be possible to add an option in the advanced part of the settings that would invalidate the whole quest cache.

I think this is tracked in #300

**Comment by westnordost** [comment]
Hmm right.

**Comment by althio** [comment]
@westnordost 
> I do not want this app to put a big load on the overpass api, which it heavily uses.

Just as a note, if needed in the future.
[Other Overpass instances ](https://wiki.openstreetmap.org/wiki/Overpass_API) are beginning to appear, especially a [new and powerful server](https://overpass.kumi.systems/api/interpreter).

**Comment by westnordost** [comment]
That'd be a good idea for the future: balance the load on different servers, depending on their geolocation. I.e. in Europe, the main instance is good but perhaps on other continents, other servers will be easier to reach.

**Comment by rugk** [comment]
Yeah, great idea. Maybe open a new issue for that?

**Comment by westnordost** [comment]
Well, I checked, there are no servers outside Europe, the main ones actually only Germany.

Server | Location
------------ | -------------
kumi.systems | Munic (Germany)
lz4.overpass-api | Nuremburg (Germany)
z.overpass-api | Berlin (Germany)
rambler | Moscow (Russia)
french | Roubaix (France)
swiss | Bern (Switzerland)
ireland | Waterford (Ireland)
